packages/parser/limits.go

```diff
@@ -0,0 +1,247 @@
+	log.WithFields(log.Fields{"type": consts.BlockError, "error": err}).Error(limitName)
+	return script.SetVMError(`panic`, err)
+}
+
+// Checking the max tx in the block
+type txMaxLimit struct {
+	Count int // the current count
+	Limit int // max count of tx in the block
+}
+
+func (bl *txMaxLimit) initLimit(b *Block) {
+	bl.Limit = syspar.GetMaxTxCount()
+}
+
+func (bl *txMaxLimit) checkLimit(p *Parser, mode int) error {
+	bl.Count++
+	if bl.Count > bl.Limit {
+		if mode == letPreprocess {
+			return ErrLimitStop
+		}
+		return limitError(`txMaxLimit`, `Max tx in the block`)
+	}
+	return nil
+}
+
+// Checking the time of the start of generating block
+type timeBlockLimit struct {
+	Start int64 // the time of the start of generating block
+	Limit int64 // the maximum time
+}
+
+func (bl *timeBlockLimit) initLimit(b *Block) {
+	bl.Start = time.Now().Unix()
+	bl.Limit = syspar.GetMaxBlockGenerationTime()
+}
+
+func (bl *timeBlockLimit) checkLimit(p *Parser, mode int) error {
+	if time.Now().Unix() > bl.Start+bl.Limit {
+		return ErrLimitStop
+	}
+	return nil
+}
+
+// Checking the max tx from one user in the block
+type txUserLimit struct {
+	TxUsers map[int64]int // the counter of tx from one user
+	Limit   int           // the value of max tx from one user
+}
+
+func (bl *txUserLimit) initLimit(b *Block) {
+	bl.TxUsers = make(map[int64]int)
+	bl.Limit = syspar.GetMaxBlockUserTx()
+}
+
+func (bl *txUserLimit) checkLimit(p *Parser, mode int) error {
+	var (
+		count int
+		ok    bool
+	)
+	keyID := p.TxSmart.KeyID
+	if count, ok = bl.TxUsers[keyID]; ok {
+		if count+1 > bl.Limit {
+			if mode == letPreprocess {
+				return ErrLimitSkip
+			}
+			return limitError(`txUserLimit`, `Max tx from one user %d`, keyID)
+		}
+	}
+	bl.TxUsers[keyID] = count + 1
+	return nil
+}
+
+// Checking the max tx from one user in the ecosystem contracts
+type ecosysLimit struct {
+	TxUsers map[int64]int // the counter of tx from one user in the ecosystem
+	Limit   int           // the value of max tx from one user in the ecosystem
+}
+
+type txUserEcosysLimit struct {
+	TxEcosys map[int64]ecosysLimit // the counter of tx from one user in ecosystems
+}
+
+func (bl *txUserEcosysLimit) initLimit(b *Block) {
+	bl.TxEcosys = make(map[int64]ecosysLimit)
+}
+
+func (bl *txUserEcosysLimit) checkLimit(p *Parser, mode int) error {
+	keyID := p.TxSmart.KeyID
+	ecosystemID := p.TxSmart.EcosystemID
+	if val, ok := bl.TxEcosys[ecosystemID]; ok {
+		if user, ok := val.TxUsers[keyID]; ok {
+			if user+1 > val.Limit {
+				if mode == letPreprocess {
+					return ErrLimitSkip
+				}
+				return limitError(`txUserEcosysLimit`, `Max tx from one user %d in ecosystem %d`,
+					keyID, ecosystemID)
+			}
+			val.TxUsers[keyID] = user + 1
+		} else {
+			val.TxUsers[keyID] = 1
+		}
+	} else {
+		limit := syspar.GetMaxBlockUserTx()
+		sp := &model.StateParameter{}
+		sp.SetTablePrefix(converter.Int64ToStr(ecosystemID))
+		found, err := sp.Get(nil, `max_block_user_tx`)
```

Comment:
better use parser transaction, because of very hard to find bugs later, like something not found, because it inserted only in transaction